include/numerics/petsc_matrix.h

```diff
@@ -163,8 +168,25 @@ class PetscMatrix final : public SparseMatrix<T>
 
   virtual numeric_index_type m () const override;
 
+  /**
+   * Get the number of rows owned by this process
+   */
+  numeric_index_type local_m () const;
```

Comment:
Can we bump this up to the parent class and make the default implementation ```row_stop()-row_start()```?

Reply:
We could, but `getting m` directly from Mat looks better for me.